java/code/src/com/redhat/rhn/frontend/xmlrpc/channel/software/ChannelSoftwareHandler.java

```diff
@@ -1702,7 +1697,8 @@ public int clone(User loggedInUser, String originalLabel,
 
         Channel originalChan = lookupChannelByLabel(loggedInUser.getOrg(), originalLabel);
 
-        CloneChannelCommand ccc = new CloneChannelCommand(originalState ? ORIGINAL_STATE : CURRENT_STATE, originalChan);
+        CloneChannelCommand ccc = new CloneChannelCommand(
+                Boolean.TRUE.equals(originalState) ? ORIGINAL_STATE : CURRENT_STATE, originalChan);
```

Comment:
Can't we just change `originalState` to `bool` rather than using the object version?

Reply:
I think not.
Reason: this is XMLRPC interface and it works only with Object types correctly when try to match the parameters.